File: backend/src/main/java/com/healthcare/appointment/model/Appointment.java
package com.healthcare.appointment.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "appointments")
@Data
@NoArgsConstructor
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private Long userId;
    
    @Column(nullable = false)
    private String doctorName;
    
    @Column(nullable = false)
    private LocalDateTime appointmentDateTime;
    
    private String reason;
    
    private String status; // SCHEDULED, CANCELLED, COMPLETED
    
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) {
            status = "SCHEDULED";
        }
    }
    
    // Getters (Lombok @Data should generate these, but adding explicitly for compatibility)
    public Long getId() { return id; }
    public Long getUserId() { return userId; }
    public String getDoctorName() { return doctorName; }
    public LocalDateTime getAppointmentDateTime() { return appointmentDateTime; }
    public String getReason() { return reason; }
    public String getStatus() { return status; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    
    // Setters
    public void setId(Long id) { this.id = id; }
    public void setUserId(Long userId) { this.userId = userId; }
    public void setDoctorName(String doctorName) { this.doctorName = doctorName; }
    public void setAppointmentDateTime(LocalDateTime appointmentDateTime) { this.appointmentDateTime = appointmentDateTime; }
    public void setReason(String reason) { this.reason = reason; }
    public void setStatus(String status) { this.status = status; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
}

File: mobile_app/lib/models/appointment.dart
class Appointment {
  final int? id;
  final int userId;
  final String doctorName;
  final DateTime appointmentDateTime;
  final String? reason;
  final String status;
  final DateTime? createdAt;

  Appointment({
    this.id,
    required this.userId,
    required this.doctorName,
    required this.appointmentDateTime,
    this.reason,
    required this.status,
    this.createdAt,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] as int?,
      userId: json['userId'] as int,
      doctorName: json['doctorName'] as String,
      appointmentDateTime: DateTime.parse(json['appointmentDateTime'] as String),
      reason: json['reason'] as String?,
      status: json['status'] as String,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'doctorName': doctorName,
      'appointmentDateTime': appointmentDateTime.toIso8601String(),
      'reason': reason,
    };
  }
}

